infrastructure/httpapi/handlers/validators.go

```diff
@@ -298,6 +311,47 @@ func (handler *Validators) getAverageBlockTime() (*big.Float, error) {
 		big.NewFloat(1000),
 	)
 
+	// totalBlockCount > nRecentBlocks: then re-calculate averageBlockTime
+	// totalBlockCount <= nRecentBlocks: use the above calculated averageBlockTime
+	nRecentBlocks := big.NewInt(nRecentBlocksInInt)
+	if totalBlockCount.Cmp(nRecentBlocks) == 1 {
+
+		latestBlockHeight, err := handler.blockView.Count()
+		if err != nil {
+			return nil, fmt.Errorf("error fetching latest block height: %v", err)
+		}
+		latestBlockIdentity := block_view.BlockIdentity{MaybeHeight: &latestBlockHeight}
+		latestBlock, err := handler.blockView.FindBy(&latestBlockIdentity)
+		if err != nil {
+			return nil, fmt.Errorf("error fetching latest block: %v", err)
+		}
+
+		// Find the nth block before latest block
+		startBlockHeight := latestBlockHeight - nRecentBlocks.Int64()
+		startBlockIdentity := block_view.BlockIdentity{MaybeHeight: &startBlockHeight}
+		startBlock, err := handler.blockView.FindBy(&startBlockIdentity)
+		if err != nil {
+			return nil, fmt.Errorf("error fetching the start block: %v", err)
+		}
+
+		// Calculate total time in generating n recent blocks
+		nRecentBlocksTotalTime := latestBlock.Time.UnixNano() - startBlock.Time.UnixNano()
+
+		nRecentBlocksTotalTimeMilliSecond := new(big.Float).Quo(
+			new(big.Float).SetInt64(nRecentBlocksTotalTime),
+			new(big.Float).SetInt64(int64(1000000)),
+		)
+		averageBlockTimeMilliSecond = new(big.Float).Quo(
+			nRecentBlocksTotalTimeMilliSecond,
+			new(big.Float).SetInt(nRecentBlocks),
+		)
+		averageBlockTime = new(big.Float).Quo(
+			averageBlockTimeMilliSecond,
+			big.NewFloat(1000),
+		)
+
+	}
```

Comment:
we can move the existing averageBlockTime calculation (using total block time) into the else part so to avoid double calculation

Reply:
Good catch! Thanks for the advice! Will update this part!